packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panel_title_card.dart

```diff
@@ -13,19 +13,23 @@ class KnowledgePanelTitleCard extends StatelessWidget {
 
   final TitleElement knowledgePanelTitleElement;
   final Evaluation? evaluation;
-
+  
   @override
   Widget build(BuildContext context) {
     final ThemeData themeData = Theme.of(context);
+    final ColorScheme colorScheme = themeData.colorScheme;
     Color? colorFromEvaluation;
+    Color? iconColorFromEvaluation;
     if (evaluation != null &&
         (knowledgePanelTitleElement.iconColorFromEvaluation ?? false)) {
       colorFromEvaluation = _getColorFromEvaluation(evaluation!);
+      iconColorFromEvaluation = _getColorFromEvaluation(evaluation!);
     }
     if (evaluation != null &&
         _getColorFromEvaluation(evaluation!) == null &&
         themeData.brightness == Brightness.dark) {
       colorFromEvaluation = Colors.white;
+      iconColorFromEvaluation = colorScheme.primary;
     }
```

Comment:
If you want my opinion:
```dart
Color? colorFromEvaluation;
if (evaluation != null &&
    (knowledgePanelTitleElement.iconColorFromEvaluation ?? false)) {
  colorFromEvaluation = _getColorFromEvaluation(evaluation!);
}
if (colorFromEvaluation != null) {
  colorFromEvaluation = colorScheme.onBackground;
}
```

Reply:
Thanks for the suggestion, it reduces an extra variable.